### PR TITLE
[18250] Documentation CI

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**

### CODE REVIEW

File: docs/test/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+endif()
+
+# Check docs style
+find_program(DOC8 doc8 DOC "path to doc8 executable")
+message(STATUS "doc8 found: ${DOC8}")
+if(DOC8)
+    add_test(
+        NAME documentation.doc8
+        COMMAND ${DOC8}
+        --ignore D001 ${PROJECT_SOURCE_DOCS_DIR})
```
Comment (JesusPoderoso) [inline docs/test/CMakeLists.txt:34]: thanks for that! 😆

File: docs/test/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+####################################################################################################
+# Add tests
+####################################################################################################
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/grep.py
+    ${CMAKE_CURRENT_BINARY_DIR}/grep.py COPYONLY)
+
+find_package(Python COMPONENTS Interpreter Development)
+if(NOT(${Python_FOUND} AND (${Python_VERSION} VERSION_GREATER "3.6")))
+    message(FATAL_ERROR "Building the documentation tests requires Python 3.7")
```
Comment (JesusPoderoso) [inline docs/test/CMakeLists.txt:24]: Shouldn't this assert or finish test execution?

File: docs/test/grep.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
+# Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (JesusPoderoso) [inline docs/test/grep.py:1]:
```suggestion
# Copyright 2023 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]

File: docs/test/CMakeLists.txt (hunk 2 above)
Comment (JLBuenoLopez-eProsima) [reply docs/test/CMakeLists.txt:24]: This is coming from other projects :sweat_smile: But looking at [CMake documentation](https://cmake.org/cmake/help/v3.0/command/message.html) when the `mode` is `FATAL_ERROR`: `CMake Error, stop processing and generation`. So it seems like it is already doing what you suggest. :wink:

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Pull request merged**